gnovm/stdlibs/std/coins.gno

```diff
@@ -13,21 +15,77 @@ type Coin struct {
 	Amount int64  `json:"amount"`
 }
 
+// NewCoin returns a new coin with a denomination and amount
+func NewCoin(denom string, amount int64) Coin {
+	return Coin{
+		Denom:  denom,
+		Amount: amount,
+	}
+}
+
+// String provides a human-readable representation of a coin
 func (c Coin) String() string {
+	if c.IsZero() {
+		return ""
+	}
+
 	return strconv.Itoa(int(c.Amount)) + c.Denom
 }
 
+// IsGTE returns true if they are the same type and the receiver is
+// an equal or greater value
 func (c Coin) IsGTE(other Coin) bool {
 	if c.Denom != other.Denom {
-		panic("invalid coin denominations: " + c.Denom)
+		panic("comparing different denominations: " + c.Denom + ", " + other.Denom)
 	}
 	return c.Amount >= other.Amount
 }
 
+// IsLT returns true if they are the same type and the receiver is
+// a smaller value
+func (c Coin) IsLT(other Coin) bool {
+	if c.Denom != other.Denom {
+		panic("comparing different denominations: " + c.Denom + ", " + other.Denom)
+	}
+	return c.Amount < other.Amount
+}
+
+// IsEqual returns true if the two sets of Coins have the same value
+func (c Coin) IsEqual(other Coin) bool {
+	if c.Denom != other.Denom {
+		panic("comparing different denominations: " + c.Denom + ", " + other.Denom)
+	}
+	return c.Amount == other.Amount
+}
+
+// IsZero returns if this represents no money
+func (c Coin) IsZero() bool {
+	return c.Amount == 0
+}
+
+func NewCoins(val ...interface{}) Coins {
```

Comment:
Why not just:

```suggestion
func NewCoins(coins ...Coin) Coins {
```

intended to be used like:

```go
NewCoins(NewCoin(20, "DENOM"), NewCoin(1, "DENOM2"))
```

I don't see the benefits of this constructor.

Reply:
Mainly because the actual calls to this would be 

`std.NewCoins(std.NewCoin(20, "ugnot"), std.NewCoins(20,"xgnot")`

which is more or less the same as, or even more complicated to write than

`std.Coins{std.Coin{20, "ugnot"), std.Coin{20,"xgnot"}}`'

The main benefit is that it is very simple to write. WDYT?

Reply:
first of all, defining structs without specifying the types is not recommended, so:

`std.Coins{std.Coin{20, "ugnot"}, std.Coin{20,"xgnot"}}` should be `std.Coins{std.Coin{Amount: 20, Denom: "ugnot"), std.Coin{Amount: 20, Denom: "xgnot"}}`

Now constructors make more sense.

Also, using types in a strongly typed language avoids errors. Using interfaces that need to have specific types in specific positions on a specific number if not will panic, look like a source of problems and bugs.

Reply:
We talked about this and we agreed to have two methods: the standard constructor using types, and a specific utility method called `CoinsFromValues(vals ... interface{})` or similar.